ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/V2.hs

```diff
@@ -757,7 +758,7 @@ newForker h ldbEnv rr st = do
   let tr = LedgerDBForkerEvent . TraceForkerEventWithKey forkerKey >$< ldbTracer ldbEnv
   traceWith tr ForkerOpen
   lseqVar <- newTVarIO . LedgerSeq . AS.Empty $ st
-  (k, toRelease) <- allocate rr (\_ -> newTVarIO (pure ())) (readTVarIO Monad.>=> id)
+  (k, toRelease) <- allocate rr (\_ -> newTVarIO (close (tables st))) (Monad.join . readTVarIO)
```

Comment:
Just to make sure: The `close` here actually also isn't strictly necessary, as it will also be closed by the `ResourceRegistry`; just like in `implForkerCommit`, right?

The LedgerDB state machine test is failing if I change this back to `pure ()`, we *don't* use a short-lived `ResourceRegistry` for `OpenAndCloseForker`.